augur/generate.py

```diff
@@ -133,15 +150,43 @@ def two_point_template(config):
             print("Missing Nz_type in %s. Quitting." % src)
             raise RuntimeError
         if tcfg["Nz_type"] == "Gaussian":
-            mu, sig = tcfg["Nz_center"], tcfg["Nz_sigma"]
-            zar = np.linspace(max(0, mu - 5 * sig), mu + 5 * sig, 500)
-            Nz = np.exp(-((zar - mu) ** 2) / (2 * sig ** 2))
-            S.add_tracer("NZ", src, zar, Nz)
+            mu, wi = tcfg["Nz_center"], tcfg["Nz_width"]
+            alpha, z0 = tcfg["Nz_alpha"], tcfg["Nz_z0"]
+            sig_z = tcfg["Nz_sigmaz"]
+            zall = np.linspace(0, 4, 1500)
+            mask = (zall > mu-wi/2) & (zall < mu+wi/2)
+            dndz_bin = np.zeros_like(zall)
+            dndz_bin[mask] = srd_dndz(zall[mask], z0, alpha)
+            # Convolve the SRD N(z) with a Gaussian with the required smearing
+            Nz = np.array([np.exp(-0.5*(zk-zall)**2/((sig_z*(1+zk))**2))*dndz_bin for zk in zall])
+            Nz = np.sum(Nz, axis=1)
+            S.add_tracer("NZ", src, zall, Nz)
+            zmean[src] = np.average(zall, weights=Nz/np.sum(Nz))
         elif tcfg["Nz_type"] == "TopHat":
             mu, wi = tcfg["Nz_center"], tcfg["Nz_width"]
+            alpha, z0 = tcfg["Nz_alpha"], tcfg["Nz_z0"]
             zar = np.linspace(max(0, mu - wi / 2), mu + wi / 2, 5)
-            Nz = np.ones(5)
+            Nz = np.ones(5)*srd_dndz(zar, z0, alpha)
             S.add_tracer("NZ", src, zar, Nz)
+            zmean[src] = np.average(zar, weights=Nz/np.sum(Nz))
+        elif tcfg["Nz_type"] == 'ConstantDensGauss':
```

Comment:
I think this would also be "ConstantDensSRD2018" or something by the suggested naming convention above.